qiskit/circuit/measure.py

```diff
@@ -59,3 +61,53 @@ def measure(self, qubit, cbit):
 
 
 QuantumCircuit.measure = measure
+
+
+class PauliMeasure(Measure):
+    """Perform a measurement with preceding basis change operations."""
+
+    def __init__(self, basis='z'):
+        """Create a new basis transformation measurement.
+
+        Args:
+            basis (str): The target measurement basis, can be 'x', 'y' or 'z'.
+
+        Raises:
+            ValueError: If an unsupported basis is specified.
+        """
+        super().__init__()
+
+        if basis.lower() == 'x':
+            self.name = 'x_measure'
+            from qiskit.extensions.standard import HGate
+            basis_transformation = [HGate()]
+        elif basis.lower() == 'y':
+            self.name = 'y_measure'
+            from qiskit.extensions.standard import RXGate
+            basis_transformation = [RXGate(pi / 2)]
+        elif basis.lower() == 'z':
+            # keep the name 'measure'
+            basis_transformation = []
+        else:
+            raise ValueError('Unsupported measurement basis choose either of x, y or z.')
+
+        self.basis = basis
+        self.basis_transformation = basis_transformation
+
+    def _define(self):
+        definition = []
+        q = QuantumRegister(1, 'q')
+        c = ClassicalRegister(1, 'c')
+        for gate in self.basis_transformation:
+            definition.append((gate, [q[0]], []))
+        definition.append((Measure(), [q[0]], [c[0]]))
+
+        self.definition = definition
+
+
+def pauli_measure(self, qubit, cbit, basis='z'):
```

Comment:
Basis should come first as a required parameter, not a kwarg (like for other standard gates `u1(theta, q)` etc.